### PR TITLE
Add logging to TRON

### LINKED ISSUE #57: Add logging to TRON

### PR DESCRIPTION
**Pull request by abelsiqueira**
Closes #57

### CODE REVIEW

File: src/solver/tron.jl [hunk 1]
```diff
@@ -65,10 +66,12 @@ function tron(nlp :: AbstractNLPModel;
 
   αC = 1.0
   tr = TRONTrustRegion(min(max(1.0, 0.1 * norm(πx)), 100))
-  if verbose
-    @printf("%4s  %9s  %7s  %7s  %7s  %s\n", "Iter", "f", "π", "Radius", "Ratio", "CG-status")
-    @printf("%4d  %9.2e  %7.1e  %7.1e\n", iter, fx, πx, get_property(tr, :radius))
-  end
+  @info(tronlogger,
+        @sprintf("%4s  %9s  %7s  %7s  %7s  %s\n",
```
Comment (dpo) [inline src/solver/tron.jl]: The `\n` should be removed. The logger automatically appends one.

File: src/solver/tron.jl [hunk 2]
```diff
@@ -229,10 +236,9 @@ function cauchy{T <: Real}(x::AbstractVector{T},
       end
       # TODO: Correctly assess why this fails
       if α < sqrt(nextfloat(zero(α)))
-        throw("α too small (qs = $qs, slope = $slope)")
-        #stalled = true
-        #status = "α too small"
-        #break
+        stalled = true
```
Comment (dpo) [inline src/solver/tron.jl]: This warrants a log message I think, at the warning or error level.

File: src/solver/tron.jl [hunk 3]
```diff
@@ -61,23 +62,31 @@ function tron(nlp :: AbstractNLPModel;
...
     xc .= x
     fc = fx
     Δ = get_property(tr, :radius)
     H = hess_op!(nlp, xc, temp)
 
-    αC, s = cauchy(x, H, gx, Δ, αC, ℓ, u, μ₀=μ₀, μ₁=μ₁, σ=σ)
+    αC, s, cauchy_status = cauchy(x, H, gx, Δ, αC, ℓ, u, μ₀=μ₀, μ₁=μ₁, σ=σ)
 
+    if cauchy_status != "success"
+      @critical(tronlogger, status)
```
Comment (dpo) [inline src/solver/tron.jl]: Maybe a little more detail here?

**Comment by dpo** [comment]
There's still a `try ... catch` block related to the ratio of achieved vs. predicted decrease.

File: src/solver/tron.jl (hunk 2 above)
Comment (abelsiqueira) [reply src/solver/tron.jl]: I ended up returning the Cauchy status to the main function and issuing a `@critical` message there. Is it better to do it here?

Comment (dpo) [reply src/solver/tron.jl]: Oh I see. Sorry. That's all good. Thanks.

**Comment by abelsiqueira** [comment]
I think I'll rebase this on top of #54 to use the new `aredpred`. Either way I'll have to rebase

**Comment by abelsiqueira** [comment]
This is on top of #54 now, so I'll have to rebase after that.

**Comment by abelsiqueira** [comment]
Rebased and includes #60

**Pull request merged**